rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/filter/TbAssetTypeSwitchNode.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright © 2016-2022 The Thingsboard Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.rule.engine.filter;
+
+import lombok.extern.slf4j.Slf4j;
+import org.thingsboard.rule.engine.api.EmptyNodeConfiguration;
+import org.thingsboard.rule.engine.api.RuleNode;
+import org.thingsboard.rule.engine.api.TbContext;
+import org.thingsboard.server.common.data.EntityType;
+import org.thingsboard.server.common.data.id.AssetId;
+import org.thingsboard.server.common.data.id.EntityId;
+import org.thingsboard.server.common.data.plugin.ComponentType;
+
+@Slf4j
+@RuleNode(
+        type = ComponentType.FILTER,
+        name = "asset type switch",
+        customRelations = true,
+        relationTypes = {},
+        configClazz = EmptyNodeConfiguration.class,
+        nodeDescription = "Route incoming messages by Asset Type",
+        nodeDetails = "Routes messages to chain according to the asset type",
+        uiResources = {"static/rulenode/rulenode-core-config.js"},
+        configDirective = "tbNodeEmptyConfig")
+public class TbAssetTypeSwitchNode extends TbAbstractTypeSwitchNode {
+
+    protected String getRelationType(TbContext ctx, EntityId originator) {
+        if (!EntityType.ASSET.equals(originator.getEntityType())) {
+            throw new RuntimeException("Unsupported originator type: " + originator.getEntityType() + "!");
+        }
+        return ctx.getAssetProfileCache().get(ctx.getTenantId(), (AssetId) originator).getName();
```

Comment:
There might be a case that the rule node will receive a msg for the entity that was removed, as a result the NPE will be thrown due to .getName() call. Same issue with DeviceTypeSwitchNode